plugin/pxr/maya/lib/usdMaya/importCommand.cpp

```diff
@@ -17,253 +17,30 @@
-    for (unsigned int i=0; i < argData.numberOfFlagUses("variant"); ++i)
-    {
-        MArgList tmpArgList;
-        status = argData.getFlagArgumentList("variant", i, tmpArgList);
-        // Get the value
-        MString tmpKey = tmpArgList.asString(0, &status);
-        MString tmpVal = tmpArgList.asString(1, &status);
-        mVariants.insert( std::pair<std::string, std::string>(tmpKey.asChar(), tmpVal.asChar()) );
-    }
-
-    bool readAnimData = true;
-    if (argData.isFlagSet("readAnimData")) {
-        argData.getFlagArgument("readAnimData", 0, readAnimData);
-    }
-
-    GfInterval timeInterval;
-    if (readAnimData) {
-        if (argData.isFlagSet("frameRange")) {
-            double startTime = 1.0;
-            double endTime = 1.0;
-            argData.getFlagArgument("frameRange", 0, startTime);
-            argData.getFlagArgument("frameRange", 1, endTime);
-            timeInterval = GfInterval(startTime, endTime);
-        }
-        else {
-            timeInterval = GfInterval::GetFullInterval();
-        }
-    }
-    else {
-        timeInterval = GfInterval();
-    }
-
-    // Create the command
-    if (mUsdReadJob) {
-        delete mUsdReadJob;
-    }
-
-    UsdMayaJobImportArgs jobArgs =
-            UsdMayaJobImportArgs::CreateFromDictionary(
-                userArgs,
-                /* importWithProxyShapes = */ false,
-                timeInterval);
-
-    MayaUsd::ImportData importData(mFileName);
-    importData.setRootVariantSelections(std::move(mVariants));
-    importData.setRootPrimPath(mPrimPath);
-
-    mUsdReadJob = new UsdMaya_ReadJobWithSceneAssembly(importData, jobArgs);
-
-    // Add optional command params
-    if (argData.isFlagSet("parent")) {
-        // Get the value
-        MString tmpVal;
-        argData.getFlagArgument("parent", 0, tmpVal);
-
-        if (tmpVal.length()) {
-            MSelectionList selList;
-            selList.add(tmpVal);
-            MDagPath dagPath;
-            status = selList.getDagPath(0, dagPath);
-            if (status != MS::kSuccess) {
-                TF_RUNTIME_ERROR(
-                        "Invalid path '%s' for -parent.",
-                        tmpVal.asChar());
-                return MS::kFailure;
-            }
-            mUsdReadJob->SetMayaRootDagPath( dagPath );
-        }
-    }
-
-    // Execute the command
-    std::vector<MDagPath> addedDagPaths;
-    bool success = mUsdReadJob->Read(&addedDagPaths);
-    if (success) {
-        TF_FOR_ALL(iter, addedDagPaths) {
-            appendToResult(iter->fullPathName());
-        }
-    }
-    return (success) ? MS::kSuccess : MS::kFailure;
-}
-
-/* virtual */
-MStatus
-UsdMayaImportCommand::redoIt()
-{
-    if (!mUsdReadJob) {
-        return MS::kFailure;
-    }
-
-    bool success = mUsdReadJob->Redo();
-
-    return (success) ? MS::kSuccess : MS::kFailure;
+    return std::unique_ptr<UsdMaya_ReadJob>(new UsdMaya_ReadJobWithSceneAssembly(data, args));
```

Comment:
Overriding the function that initializes the read job and returning the UsdMaya_ReadJobWithSceneAssembly that the Pixar plugin uses to customize the import behavior.